### PR TITLE
TRAD-334 Amm Support for Submitting TradeSwap/RouteTradeSwap

### PR DESCRIPTION
**Pull request by kdonthi**
* Creating functions `submitTradeSwap` + `submitRouteTradeSwap`
* Added support for submitting by batch (`postSubmitBatch`)
* Handling errors from all HTTP endpoints

### CODE REVIEW

File: bxsolana/provider/base.ts [hunk 1]
```diff
@@ -233,6 +235,44 @@ export abstract class BaseProvider implements Api {
...
+    async submitTradeSwap(request: TradeSwapRequest, submitStrategy: SubmitStrategy, skipPreFlight = true): Promise<PostSubmitBatchResponse> {
+        const res = await this.postTradeSwap(request)
+
+        const entries = new Array<PostSubmitRequestEntry>()
+        for (const transactionMessage of res.transactions) {
+            entries.push({
+                transaction: signTxMessage(transactionMessage),
+                skipPreFlight: skipPreFlight,
+            })
+        }
```
Comment (aspin) [inline bxsolana/provider/base.ts]: add a method for this for reuse. `submitTransactions`?

Comment (aspin) [reply bxsolana/provider/base.ts]: probably could consider this for the singular ones as well

File: bxsolana/provider/http.ts [hunk 2]
```diff
@@ -43,6 +45,7 @@ import {
 } from "../proto/messages/api/index.js"
 import { BaseProvider } from "./base.js"
 import config from "../../utils/config.js"
+import { IsRPCError, RPCError } from "./error.js"
```
Comment (aspin) [inline bxsolana/provider/http.ts]: Are you missing a file here?

BTW, naming: function should be `isRpcError`, and class should be `RpcError`. TS doesn't capitalize acronyms

File: examples/index.ts [hunk 3]
```diff
@@ -58,8 +57,8 @@ async function http() {
     const provider = new HttpProvider()
     console.info(" ----  HTTP Requests  ----")
     await doRequests(provider)
-    console.info(" ----  HTTP Solana Requests  ----")
-    await doSolanaRequests(provider)
+    console.info(" ----  HTTP Amm Requests  ----")
+    await doAmmRequests(provider)
```
Comment (aspin) [inline examples/index.ts:61]: :+1:

**Review by aspin — CHANGES_REQUESTED** [review]
think you're missing a file, but changes look good :+1: :+1:

File: bxsolana/provider/http.ts (hunk 2 above)
Comment (kdonthi) [reply bxsolana/provider/http.ts]: Yes! I didn't realize it 😅

File: utils/error.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export interface RpcError {
+    code: number
+    message: string
+}
+
+export function IsRpcError(obj: any): obj is RpcError {
```
Comment (aspin) [inline utils/error.ts]: naming => `isRpcError`. Shouldn't the return type be bool?

**Review by aspin — APPROVED** [review]

**Review by atulsriv — CHANGES_REQUESTED** [review]
Can you verify that AMM Requests and AMM streams successfully executes on examples? https://app.circleci.com/pipelines/github/bloXroute-Labs/solana-trader-client-ts/289/workflows/6981ac63-ea11-488e-a69c-207eeca1e431/jobs/353

I'm having issues running this PR on my PR with the integration tests. I have merged this branch in

**Comment by atulsriv** [comment]
> Can you verify that AMM Requests and AMM streams successfully executes on examples? https://app.circleci.com/pipelines/github/bloXroute-Labs/solana-trader-client-ts/289/workflows/6981ac63-ea11-488e-a69c-207eeca1e431/jobs/353
> 
> I'm having issues running this PR on my PR with the integration tests. I have merged this branch in

particularly, against mainnet

**Review by atulsriv — APPROVED** [review]

**Pull request merged**